src/components/ItemAccordion/ItemAccordion.test.js

```diff
@@ -41,4 +48,37 @@ describe('ItemAccordion', () => {
     expect(screen.getByRole('group', { name: /electronic access/i })).toBeInTheDocument();
     expect(screen.getByRole('button', { name: /add electronic access/i })).toBeInTheDocument();
   });
+
+  it('should not render generate barcode button and enable barcode field when settings=off', () => {
+    useData.mockReturnValue({
+      numberGeneratorSettings: { [BARCODE_SETTING]: NUMBER_GENERATOR_OPTIONS_OFF }
+    });
+
+    renderItemAccordion();
+
+    expect(screen.queryByRole('button', { name: 'Generate barcode' })).not.toBeInTheDocument();
+    expect(screen.getByRole('textbox', { name: 'Barcode' })).toBeEnabled();
+  });
+
+  it('should render generate barcode button and disable barcode field when settings=onNotEditable', () => {
+    useData.mockReturnValue({
+      numberGeneratorSettings: { [BARCODE_SETTING]: NUMBER_GENERATOR_OPTIONS_ON_NOT_EDITABLE }
+    });
+
+    renderItemAccordion();
+
+    expect(screen.getByRole('button', { name: 'Generate barcode' })).toBeInTheDocument();
+    expect(screen.getByRole('textbox', { name: 'Barcode' })).toBeDisabled();
+  });
```

Comment:
I understand this _test_ works. But a disabled field won't be part of the form submission, will it? e.g. https://codesandbox.io/p/sandbox/7ngw4h

Reply:
The behavior of `react-final-form` differs from the native HTML form
You are right, in a classic HTML form with disabled, the value would not be sent.
But a react-final-form sends all saved values, regardless of whether the corresponding input field is disabled or not.